lib/perso.ml

```diff
@@ -5328,8 +5328,21 @@ let eval_predefined_apply conf env f vl =
       | _ -> raise Not_found)
   | "min", s :: sl -> (
       try
+        let s0 = if s = "" then max_int else int_of_string s in
         let m =
-          List.fold_right (fun s -> min (int_of_string s)) sl (int_of_string s)
+          List.fold_right
+            (fun s -> min (if s = "" then max_int else int_of_string s))
+            sl s0
+        in
+        string_of_int m
+      with Failure _ -> raise Not_found)
+  | "max", s :: sl -> (
+      try
+        let s0 = if s = "" then 0 - max_int else int_of_string s in
+        let m =
+          List.fold_right
+            (fun s -> max (if s = "" then 0 - max_int else int_of_string s))
+            sl s0
         in
```

Comment:
looks good to me,

for clarity I would do the handling the s = "" case and string to int conversion before the fold, maybe like this:
```ocaml
  | "max", l -> (
      try
        let l = List.map (fun s -> if s = "" then 0 - max_int else int_of_string s ) l in
        let m = List.fold_left max 0 l in
...
```

Reply:
how about: 
```
  | "max", sl -> (
      try
        let sl = List.map (fun s -> if s = "" then 0 - max_int else int_of_string s ) sl in
        let m = List.fold_right (fun s -> max s) sl (0 - max_int) in
        string_of_int m
      with Failure _ -> raise Not_found)

```

Reply:
It is better to always use fold_left because fold_right is not tail-recursive.
Even on a small list I think it is better for code consistency.

Reply:
got it! done

Reply:
thanks